### ISSUE TITLE
DuckDNS add-on fails when using adguard (or other pihole)

### ISSUE THREAD

**Issue by kodsama** [home-assistant/hassio-addons#1488]
Hi all,

This is basically  the same as #320 and HA Core issues #11699 and #22024.

When running pihole or adguard on the network (in my case adguard addon on the same device) it seems that the duckdns addon can't resolve the letsecnrypt address and silently shuts down.

I can confirm that this is the source of the problem, deactivating the adguard DNS resolving on my network allows duckDNS to correctly finish.

The only solution I could read which would solve this is to provide the duckdns addon with a parameter to use a custom DNS resolver (such as 1.1.1.1 or other) which would prevent going through adguard/pihole and allow a correct certificate validation.

Some background on my setup:
Home Assistant release with the issue:
Version 0.112.4 (HassOS 4.11)

Last working Home Assistant release (if known):
N/A

Operating environment (Hass.io/Docker/Windows/etc.):
Hass.io

Plarform:
Raspberry Pi 3B

Component/platform:
DuckDNS 1.12.1
AdGuard Home 2.4.2

**Comment by colindouglas** [comment]
I am also able to reproduce this problem on clean install of Home Assistant (HassOS 4.11) and Pi-hole for adblocking. 

If I access through HTTP, things work fine, but going through HTTPS (using either a local IP or a DuckDNS address) gives me 'SSL_ERROR_RX_RECORD_TOO_LONG' in Firefox.

Deactivating Pi-hole resolves the problem, but once Pi-hole is re-enabled, the problem returns. The only blocked request from Home Assistant is a request to ip-api.com. Whitelisting this domain does not resolve the problem

Home Assistant release with the issue:
Version 0.112.4 (HassOS 4.11)

Last working Home Assistant release (if known):
N/A

Operating environment (Hass.io/Docker/Windows/etc.):
Hass.io

Platform:
Raspberry Pi 3B

Component/platform:
DuckDNS 1.12.1
Pi-hole 5.1.1, FTL v5.1 (on a separate Pi0W)

**Comment by kodsama** [comment]
> SSL_ERROR_RX_RECORD_TOO_LONG

@colindouglas, did you try the fix described [https://community.home-assistant.io/t/duckdns-and-ssl-problem-ssl-error-rx-record-too-long/48820/13](here) ?
I had this before and it is because I needed to set the address in the http part of configuration.yaml

**Comment by colindouglas** [comment]
@kodsama I am conscious of the fix you described, and have tried it, but I don't think it is the problem here because:

1) I am using the DuckDNS add on, which doesn't require any changes to to my configuration.yaml files
2) Troubleshooting this problem and searching for an errant space has lead me to do a clean installation of HassOS, so my config files are all empty at this point
3) The problem resolves itself when Pi-hole is disabled

**Comment by kodsama** [comment]
Ok, just double checking @colindouglas (you never know)

**Comment by colindouglas** [comment]
An update, I may have jumped the gun and my problem may be unrelated. A fresh install of Pi-hole resolved the issue, although I'm not sure what it was.

**Comment by ludeeus** [comment]
So this issue can now be closed @colindouglas ?

**Comment by kodsama** [comment]
Well, I still see the problem on my setup @ludeeus 

I get that log from the duckDNS add-on when I have adguard running as DNS on my network:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
# INFO: Using main config file /data/workdir/config
ERROR: Problem connecting to server (get for https://acme-v02.api.letsencrypt.org/directory; curl returned with 6)
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

**Comment by ludeeus** [comment]
Error 6 for curl tells that you have DNS resolve issues on you installation

**Comment by zbeez** [comment]
Hi there, well I bumped into the exact same issue with Adguard Home now. I've set the DNS IP addresses of my Pi (Adguard Home) on the router and had exactly the same behaviour as @kodsama described after approx. 1,5 -2 month on a reboot of my Home Assistant. I searched for a specific solution, but the only way to tackle with that on a short term is to enter the DNS IP addresses from a DNS provider, reboot Home Assistant and revert back. 
The fullchain.pem file and privkey.pem file from root../ssl will be updated and from this point on I'm good for another 1,5-2 month. 
Is there any other way to solve this in a more clean/automated solution?
